script.js

```diff
@@ -1,12 +1,360 @@
+    episodeSummary.innerHTML = `${episode.summary}`;
+
+    eachEpisodeContainer.append(
+      title,
+      seasonEpisodeNumber,
+      imgContainer,
+      episodeSummary
+    );
+    eachEpisodeContainer.classList.add("col-3");
+    rowElem.appendChild(eachEpisodeContainer);
+    episodesContainer.appendChild(rowElem);
+  }
+  container.appendChild(episodesContainer);
+}
+
+function makeSearchForShows() {
+  const allShows = getAllShows();
+  let container = document.getElementById("root");
+
+  let searchInput = document.createElement("input");
+  searchInput.classList.add("search-input");
+  searchInput.placeholder = "Search here";
+
+  container.appendChild(searchInput);
+  searchResult.classList.add("search-result");
+
+  searchInput.addEventListener("keyup", (event) => {
+    let searchString = event.target.value.toLowerCase();
+
+    let filteredShows = allShows.filter((show) => {
+      return (
+        show.name.toLowerCase().includes(searchString) ||
+        show.summary.toLowerCase().includes(searchString)
+      );
+    });
+    searchResult.innerHTML = `Displaying ${filteredShows.length}/${allShows.length}`;
+
+    container.appendChild(searchResult);
+    document.getElementById("container").remove();
+    makePageForShows(filteredShows);
+  });
+}
+
+function makeSearchForShows() {
+  const allShows = getAllShows();
+  let container = document.getElementById("root");
+
+  let searchInput = document.createElement("input");
+  searchInput.classList.add("search-input");
+  searchInput.placeholder = "Search here";
+
+  container.appendChild(searchInput);
+  searchResult.classList.add("search-result");
+
+  searchInput.addEventListener("keyup", (event) => {
+    let searchString = event.target.value.toLowerCase();
+
+    let filteredShows = allShows.filter((show) => {
+      return (
+        show.name.toLowerCase().includes(searchString) ||
+        show.summary.toLowerCase().includes(searchString)
+      );
+    });
+    searchResult.innerHTML = `Displaying ${filteredShows.length}/${allShows.length}`;
+
+    container.appendChild(searchResult);
+    document.getElementById("container").remove();
+    makePageForShows(filteredShows);
+  });
+}
+
+function makeSearchForEpisodes() {
+  let container = document.getElementById("root");
+
+  let searchInput = document.createElement("input");
+  searchInput.classList.add("search-input");
+  searchInput.placeholder = "Search here";
+
+  container.appendChild(searchInput);
+  searchResult.classList.add("search-result");
+
+  searchInput.addEventListener("keyup", (event) => {
+    let searchString = event.target.value.toLowerCase();
+
+    let filteredEpisodes = allEpisodes.filter((show) => {
+      return (
+        episode.name.toLowerCase().includes(searchString) ||
+        episode.summary.toLowerCase().includes(searchString)
+      );
+    });
+    searchResult.innerHTML = `Displaying ${filteredShows.length}/${allShows.length}`;
```

Comment:
The `${filteredShows.length}` is not updating appropriately, perhaps because the function is not being used. Not sure if this is the only place where it happens, though... sorry, I can't be more specific here